services/agent/tests/unit/test_modules_task_monitor.py

```diff
@@ -0,0 +1,68 @@
+# pylint:disable=protected-access
+
+import asyncio
+from typing import Final, Optional
+
+import pytest
+from pydantic import PositiveFloat
+from pytest import LogCaptureFixture
+from simcore_service_agent.modules.task_monitor import TaskMonitor
+
+REPEAT_TASK_INTERVAL_S: Final[PositiveFloat] = 0.05
+
+
+async def _error_raising_job() -> None:
+    raise RuntimeError("raised expected error")
+
+
+@pytest.mark.parametrize("repeat_interval_s", [REPEAT_TASK_INTERVAL_S, None])
+async def test_task_monitor_recovers_from_error(
+    caplog_info_debug: LogCaptureFixture,
+    repeat_interval_s: Optional[PositiveFloat],
+):
+
+    task_monitor = TaskMonitor()
+    task_monitor.register_job(_error_raising_job, repeat_interval_s=repeat_interval_s)
+
+    await task_monitor.start()
+
+    await asyncio.sleep(REPEAT_TASK_INTERVAL_S * 2)
```

Comment:
you might want to add a test where the background task hangs, and then check the heatlth.

Reply:
OK, I've also managed to introduce a basic hanging task detector which will make the service unhealthy.